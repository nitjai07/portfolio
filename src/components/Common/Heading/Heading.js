import React from 'react';
import PropTypes from 'prop-types';

const Heading = props => <h1 className="commonheading" dangerouslySetInnerHTML={{ __html: props.headingText }} />;

Heading.propTypes = {
  headingText: PropTypes.string.isRequired
};

export default Heading;
