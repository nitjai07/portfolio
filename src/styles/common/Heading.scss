.commonheading {
  color: $primaryColor1;
  margin: 0;
  padding-top: 30px;
  font-size: 50px;
  font-family: 'Montserrat', sans-serif;
  font-weight: 800;
  &:before {
    content: '<h1>';
    color: $primaryColor3;
    font-size: 16px;
    position: absolute;
    font-style: italic;
    margin-top: -30px;
  }
  &:after {
    @extend :before;
    content: '<h1/>';
    margin-top: 32px;
    margin-left: 20px;

  }
}