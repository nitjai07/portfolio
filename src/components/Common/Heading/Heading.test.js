import React from 'react';
import { shallow } from 'enzyme';
import Heading from './Heading';

test('Heading Component renders correctly', () => {
  const component = shallow(<Heading headingText="test heading" />);
  expect(component).toMatchSnapshot();
});
