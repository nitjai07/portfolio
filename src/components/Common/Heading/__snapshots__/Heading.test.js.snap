// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h1
    className="commonheading"
    dangerouslySetInnerHTML={
        Object {
            "__html": "test heading",
          }
    }
/>,
  "nodes": Array [
    <h1
      className="commonheading"
      dangerouslySetInnerHTML={
            Object {
                  "__html": "test heading",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Heading
      headingText="test heading"
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <h1
      className="commonheading"
      dangerouslySetInnerHTML={
            Object {
                  "__html": "test heading",
                }
      }
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    headingText="test heading"
/>,
}
`;
